core/vm/contracts.go

```diff
@@ -358,3 +374,68 @@ func (c *bn256Pairing) Run(input []byte) ([]byte, error) {
 	}
 	return false32Byte, nil
 }
+
+type blake2F struct{}
+
+func (c *blake2F) RequiredGas(input []byte) uint64 {
+	if len(input) != 213 {
+		// Input is malformed, we can't read the number of rounds.
+		// Precompile can't be executed so we set its price to 0.
+		return 0
+	}
+
+	rounds := binary.BigEndian.Uint32(input[209:213])
+	return uint64(rounds)
+}
+
+var (
+	blake2FInputLength = 213
+
+	errBlake2FIncorrectInputLength = errors.New(
+		"input length for Blake2 F precompile should be exactly 213 bytes",
+	)
+)
+
+func (c *blake2F) Run(input []byte) ([]byte, error) {
+	if len(input) != 213 {
+		return nil, errBlake2FIncorrectInputLength
```

Comment:
If it's only used in this place, might as well define the error inline
```suggestion
		return nil, errors.New("invalid input length for Blake2 F precompile")
```

Reply:
It's also used in tests, see `TestPrecompileBlake2FMalformedInput`.